### DISCUSSION TITLE
Boolean in default annotation raise issue

### DISCUSSION THREAD

**Discussion by Smixi** [fastapi/typer#921] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typing import Annotated
app = typer.Typer()
user_app = typer.Typer()

@user_app.command("create")
def create_user(a_boolean: Annotated[bool,
        typer.Option(
            default=False,
            help="Help string stufff ....",
        )]):
    print("did something")

app.add_typer(user_app)

if __name__ == "__main__":
    app()
```

### Description

Hi, running the above script raises ```AttributeError: 'bool' object has no attribute 'isidentifier'```

This is caused by the `default=False` option in the Option. 
I use click 8.1.7, and this one works fine:

```python
import click

@click.command()
@click.option("--mybool", default=False, help="Number of greetings.")
def hello(mybool):
    print("test", )

if __name__ == '__main__':
    hello()
```

Also, when I set the default value directly as a default argument it does works :
```python
@user_app.command("create")
def create_user(a_boolean: Annotated[bool,
        typer.Option(
            help="Help string stufff ....",
        )] = False ):
    print("did something")

app.add_typer(user_app)

if __name__ == "__main__":
    app()

```
The docs says that default is supported, but it seems to works with strings, but not with bool.

### Operating System

Linux

### Operating System Details

Debian Bookworm

### Typer Version

0.12.3

### Python Version

3.11.2

**Comment by ShihaoXu** [comment]
Hey, I hope it's not that late...

I've started to use Typer recently and encountered the same issue.
You can find a (maybe) related discussion [here](https://github.com/fastapi/typer/discussions/861).

I have a fix for your issue. Instead of using `default` from within `typer.Option()`, try to set your default outside of `Annotated[]` like this:
```python
def create_user(a_boolean: Annotated[bool,
        typer.Option(
            help="Help string stufff ....",
        )] = False):
```

THe code works with the fix.

**Reply by Smixi** [reply]
Yep, this is what the original post says :D.
The issue was also that this error is weird and the docs says the opposite.
I didn't tried if this is correct yet.
